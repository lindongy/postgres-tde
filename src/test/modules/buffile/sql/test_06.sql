-- This test shows that the first component file (segment) stays empty, read
-- stops prematurely even if it starts on that segment, even though it'd
-- otherwise receive some data from the following one.
BEGIN;
SELECT buffile_create();
SELECT buffile_seek(0, 32768);
SELECT buffile_write('a');
SELECT buffile_seek(0, 32767);
SELECT buffile_read(2);
SELECT buffile_close();
COMMIT;
