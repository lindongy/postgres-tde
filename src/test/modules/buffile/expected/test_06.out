-- This test shows that the first component file (segment) stays empty, read
-- stops prematurely even if it starts on that segment, even though it'd
-- otherwise receive some data from the following one.
BEGIN;
SELECT buffile_create();
 buffile_create 
----------------
 
(1 row)

SELECT buffile_seek(0, 32768);
 buffile_seek 
--------------
            0
(1 row)

SELECT buffile_write('a');
 buffile_write 
---------------
             1
(1 row)

SELECT buffile_seek(0, 32767);
 buffile_seek 
--------------
            0
(1 row)

SELECT buffile_read(2);
 buffile_read 
--------------
 \x
(1 row)

SELECT buffile_close();
 buffile_close 
---------------
 
(1 row)

COMMIT;
